ext/pdo_firebird/firebird_driver.c

```diff
@@ -460,6 +460,51 @@ static int php_firebird_preprocess(const zend_string* sql, char* sql_out, HashTa
 	return 1;
 }
 
+#if FB_API_VER >= 40
+/* set coercing a data type */
+static void set_coercing_data_type(XSQLDA* sqlda)
+{
+    /* Data types introduced in Firebird 4.0 are difficult to process using the Firebird Legacy API. */ 
+	/* These data types include DECFLOAT(16), DECFLOAT(34), INT128 (NUMERIC/DECIMAL(38, x)), */
+	/* TIMESTAMP WITH TIME ZONE, and TIME WITH TIME ZONE. In any case, at least the first three data types */
+	/* can only be mapped to strings. The last two too, but for them it is potentially possible to set */
+	/* the display format, as is done for TIMESTAMP. This function allows you to ensure minimal performance */ 
+	/* of queries if they contain columns and parameters of the above types. */
+	unsigned int i;
+	short dtype;
+	short nullable;
+	XSQLVAR* var;
+	for (i=0, var = sqlda->sqlvar; i < sqlda->sqld; i++, var++) {
+        dtype = (var->sqltype & ~1); /* drop flag bit  */
+		nullable = (var->sqltype & 1); 
+		switch(dtype) {
+			case SQL_INT128:
+			    var->sqltype = SQL_VARYING + nullable;
+				var->sqllen = 46;
+				var->sqlscale = 0;
+			    break;
+			case SQL_DEC16:
+			    var->sqltype = SQL_VARYING + nullable;
+				var->sqllen = 24;
+			    break;
+			case SQL_DEC34:
+			    var->sqltype = SQL_VARYING + nullable;
+				var->sqllen = 43;
+			    break;
+			case SQL_TIMESTAMP_TZ:
+			    var->sqltype = SQL_VARYING + nullable;
+				var->sqllen = 58;
+			    break;
+			case SQL_TIME_TZ:
+			    var->sqltype = SQL_VARYING + nullable;
+				var->sqllen = 46;
+			    break;														
+		}
+	}	
+}
```

Comment:
Please standardize indentation to TAB.
```suggestion
static void set_coercing_data_type(XSQLDA* sqlda)
{
	/* Data types introduced in Firebird 4.0 are difficult to process using the Firebird Legacy API. */ 
	/* These data types include DECFLOAT(16), DECFLOAT(34), INT128 (NUMERIC/DECIMAL(38, x)), */
	/* TIMESTAMP WITH TIME ZONE, and TIME WITH TIME ZONE. In any case, at least the first three data types */
	/* can only be mapped to strings. The last two too, but for them it is potentially possible to set */
	/* the display format, as is done for TIMESTAMP. This function allows you to ensure minimal performance */ 
	/* of queries if they contain columns and parameters of the above types. */
	unsigned int i;
	short dtype;
	short nullable;
	XSQLVAR* var;
	for (i=0, var = sqlda->sqlvar; i < sqlda->sqld; i++, var++) {
		dtype = (var->sqltype & ~1); /* drop flag bit  */
		nullable = (var->sqltype & 1); 
		switch(dtype) {
			case SQL_INT128:
				var->sqltype = SQL_VARYING + nullable;
				var->sqllen = 46;
				var->sqlscale = 0;
				break;
			case SQL_DEC16:
				var->sqltype = SQL_VARYING + nullable;
				var->sqllen = 24;
				break;
			case SQL_DEC34:
				var->sqltype = SQL_VARYING + nullable;
				var->sqllen = 43;
				break;
			case SQL_TIMESTAMP_TZ:
				var->sqltype = SQL_VARYING + nullable;
				var->sqllen = 58;
				break;
			case SQL_TIME_TZ:
				var->sqltype = SQL_VARYING + nullable;
				var->sqllen = 46;
				break;														
		}
	}	
}
```

Reply:
Also, there is an `EMPTY_SWITCH_DEFAULT_CASE()` macro that can be used for the default case where the switch does nothing.

Reply:
If you insert this macro, the driver crashes as soon as the dtype goes beyond the cases described in case. In my case, for other types you just need to do nothing.